modular_skyrat/modules/modular_implants/code/soulcatcher/handheld_soulcatcher.dm

```diff
@@ -47,68 +56,112 @@
 		to_chat(user, span_warning("You are unable to remove a mind from an empty body."))
 		return FALSE
 
-	if(!COOLDOWN_FINISHED(src, rsd_scan_cooldown))
-		var/time_left = round((COOLDOWN_TIMELEFT(src, rsd_scan_cooldown)) / (1 MINUTES), 0.01)
-		to_chat(user, span_warning("You are currently unable to grab the soul of [target_mob], please wait [time_left] minutes before trying again."))
+	if (interacting_mobs[user])
+		to_chat(user, span_warning("You already have room selection open, close it or choose a room!"))
 		return FALSE
 
-	if(target_mob.stat == DEAD) //We can temporarily store souls of dead mobs.
-		target_mob.ghostize(TRUE) //Incase they are staying in the body.
-		var/mob/dead/observer/target_ghost = target_mob.get_ghost(TRUE, TRUE)
-		if(!target_ghost)
-			to_chat(user, span_warning("You are unable to get the soul of [target_mob]!"))
-			return FALSE
+	if (target_mob in confirming_entry)
+		to_chat(user, span_warning("[target_mob] has already been invited into this RSD! Wait for them to respond!"))
+		return FALSE
 
-		var/datum/soulcatcher_room/target_room = tgui_input_list(user, "Choose a room to send [target_mob]'s soul to.", name, linked_soulcatcher.soulcatcher_rooms, timeout = 30 SECONDS)
-		if(!target_room)
-			return FALSE
+	// dont worry about invalid states in regards to this tgui_list, it uses a custom ui_state that prevents abuse
+	interacting_mobs[user] = target_mob // prevents list spam by storing the fact we clicked on this mob
+	var/datum/soulcatcher_room/target_room = tgui_input_list(user, "Choose a room to send [target_mob]'s soul to.", name, linked_soulcatcher.soulcatcher_rooms, ui_state = new /datum/ui_state/handheld_soulcatcher_state(src), timeout = 30 SECONDS)
+	interacting_mobs -= user // tgui lists sleep - trust no variables after this point
+	if(!target_room)
+		return FALSE
 
-		SEND_SOUND(target_ghost, 'sound/misc/notice2.ogg')
-		window_flash(target_ghost.client)
+	if(!target_mob.mind)
+		to_chat(user, span_warning("You are unable to remove a mind from an empty body."))
+		return FALSE
 
-		if(tgui_alert(target_ghost, "[user] wants to transfer you to [target_room] inside of a soulcatcher, do you accept?", name, list("Yes", "No"), 30 SECONDS, autofocus = FALSE) != "Yes")
-			to_chat(user, span_warning("[target_mob] doesn't seem to want to enter."))
-			COOLDOWN_START(src, rsd_scan_cooldown, RSD_ATTEMPT_COOLDOWN)
-			return FALSE
+	to_chat(user, span_blue("[target_mob] has been requested to join [target_room]."))
+	if (!invite_soul(target_room, user, target_mob))
```

Comment:
Choose whether you put a space or not after your `if`, and stick to it, try to avoid alternating.

Reply:
i usually stick with if () but softcerv goes with if(), probably why theres some alternation, i was trying to convert some shit to no space and missed a bunch